### PR TITLE
Fix issue with X509VerificationFlags.AllowUnknownCertificateAuthority behavior

### PR DESCRIPTION
**Pull request by davidorbelian**
Fix [this issue](https://github.com/kubernetes-client/csharp/issues/173) with [this solution](https://github.com/dotnet/corefx/issues/30284).

How to test:
Replace your _certificate-authority-data_ from _kubeconfig_ with data from another certificate and run. 
- BEFORE:
The program works despite the non-validity of the certificate because of validation does not fail.
- AFTER:
**System.Net.Http.HttpRequestException** is thrown because of validation fails.

### CODE REVIEW

File: src/KubernetesClient/Kubernetes.ConfigInit.cs [hunk 1]
```diff
@@ -179,8 +180,11 @@ private void SetCredentials(KubernetesClientConfiguration config, HttpClientHand
 
                 // add all your extra certificate chain
                 chain.ChainPolicy.ExtraStore.Add(caCert);
-                chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
-                var isValid = chain.Build((X509Certificate2) certificate);
+                chain.Build((X509Certificate2) certificate);
+
+                var rootCert = chain.ChainElements[chain.ChainElements.Count - 1].Certificate;
+                var isValid = rootCert.RawData.SequenceEqual(caCert.RawData);
```
Comment (bartonjs) [inline src/KubernetesClient/Kubernetes.ConfigInit.cs]: If you want to test for a custom root you'd leave the check as it was, and then add  `isValid = isValid && rootCert.RawData.SequenceEqual(caCert.RawData);`

As it is, this change would start accepting expired and revoked certificates (and possibly certificates where the signatures didn't check out).

Comment (davidorbelian) [reply src/KubernetesClient/Kubernetes.ConfigInit.cs]: Thank you!

**Comment by davidorbelian** [comment]
@bartonjs can you please confirm now it is ok?

**Comment by davidorbelian** [comment]
@bartonjs Sure, thank you!
@brendandburns What you think?

**Comment by davidorbelian** [comment]
@brendandburns Done!

**Comment by brendandburns** [comment]
LGTM, many thanks!

**Pull request merged**